tests/acceptance/features/apiLocks/lockFiles.feature

```diff
@@ -256,3 +256,96 @@ Feature: lock files
+      | spaces           | shared     |
+      | spaces           | exclusive  |
+
+  @issue-7638 @issue-7599
+  Scenario Outline: locking a file in a received share does not lock other items with the same name in other received shares (shares from different users)
+    Given using <dav-path-version> DAV path
+    And user "Carol" has been created with default attributes and without skeleton files
+    And user "Alice" has created folder "FromAlice"
+    And user "Brian" has created folder "FromBrian"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/FromAlice/textfile0.txt"
+    And user "Brian" has uploaded file "filesForUpload/textfile.txt" to "/FromBrian/textfile0.txt"
+    And user "Alice" has shared folder "FromAlice" with user "Carol"
+    And user "Brian" has shared folder "FromBrian" with user "Carol"
+    When user "Carol" locks file "/Shares/FromBrian/textfile0.txt" using the WebDAV API setting the following properties
+      | lockscope | <lock-scope> |
+    Then the HTTP status code should be "200"
+    And user "Carol" should be able to upload file "filesForUpload/lorem.txt" to "/Shares/FromAlice/textfile0.txt"
+    But user "Carol" should not be able to upload file "filesForUpload/lorem.txt" to "/Shares/FromBrian/textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
+
+  @issue-7638 @issue-7599
+  Scenario Outline: locking a file in a received share does not lock other items with the same name in other received shares (shares from same user)
+    Given using <dav-path-version> DAV path
+    And user "Alice" has created folder "locked/"
+    And user "Alice" has created folder "notlocked/"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/locked/textfile0.txt"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/notlocked/textfile0.txt"
+    And user "Alice" has shared folder "locked" with user "Brian"
+    And user "Alice" has shared folder "notlocked" with user "Brian"
+    When user "Brian" locks file "/Shares/locked/textfile0.txt" using the WebDAV API setting the following properties
+      | lockscope | <lock-scope> |
+    Then the HTTP status code should be "200"
+    And user "Brian" should be able to upload file "filesForUpload/lorem.txt" to "/Shares/notlocked/textfile0.txt"
+    But user "Brian" should not be able to upload file "filesForUpload/lorem.txt" to "/Shares/locked/textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
```

Comment:
just interesting. what is difference between lock-scope? do we need use `shared`  lock-scope. It didn't work for me

Reply:
As per my understanding `shared` lock should i.e. read-only lock, multiple entities can read the data without updating it and exclusive lock should be write-lock, only one entity has right to both read and write data. As per the behaviour of ocis currently shared lock request is "successful" when the file is not shared, but the shared file gives error like the issue that you created https://github.com/owncloud/ocis/issues/7599, I have put them in expected to fail.